src/mod/endpoints/mod_sofia/sofia.c

```diff
@@ -1513,9 +1513,16 @@ static void our_sofia_event_callback(nua_event_t event,
 		}
 	}
 
-
 	if (sofia_private && sofia_private != &mod_sofia_globals.destroy_private && sofia_private != &mod_sofia_globals.keep_private) {
 		if (!zstr(sofia_private->gateway_name)) {
+			if (event == nua_r_unregister && (status == 200 || status == 408)) {
```

Comment:
I think the check here is that it will not perform deletion if status is either 401 or 407. What happened if for example registrar sent a 403 forbidden because the user is now been block or changed password?

Reply:
I see, I will fix the condition